rclcpp/test/topic_statistics/test_subscription_topic_statistics.cpp

```diff
@@ -97,21 +99,49 @@ class EmptyPublisher : public rclcpp::Node
 
   virtual ~EmptyPublisher() = default;
 
-  size_t get_number_published()
+private:
+  void publish_message()
+  {
+    auto msg = Empty{};
+    publisher_->publish(msg);
+  }
+
+  rclcpp::Publisher<Empty>::SharedPtr publisher_;
+  rclcpp::TimerBase::SharedPtr publish_timer_;
+};
+
+/**
+ * MessageWithHeader publisher node: used to publish MessageWithHeader with `header` value set
+ */
+class DummyPublisher : public rclcpp::Node
```

Comment:
I would renaming the new publisher/subscription nodes to something more informative, e.g. containing the name of the message: `MessageWithHeaderPublisher` and `MessageWithHeaderSubscriber`